src/components/ToDoList/ToDoList.tsx

```diff
@@ -0,0 +1,237 @@
+        if (todo.index === id) {
+          todo.isCompleted = !todo.isCompleted;
+        }
+        return todo;
+      })
+    );
+  }
+
+  function handleDelete(id: number): void {
+    setTodos(todos.filter((item) => item.index !== id));
+    if (todos.length === 1) {
+      setFilter(Filters.All)
+      setDisableStatusButton(true);
+
+    }
+  }
+
+  function handleChangeInputEditing(
+    event: React.ChangeEvent<HTMLInputElement>
+  ): void {
+    setInputValueEditing(event.target.value);
+  }
+
+  function handleEditingItem(id: number, inputTitle: string): void {
+    setEditingItem(id);
+    setInputValueEditing(inputTitle);
+  }
+
+  function handleEditingSave(id: number): void {
+    if (inputValueEditing.trim() !== '' && inputValueEditing.length <= MY_SUPER_MEGA_MAGIC_NUMBER) {
+      setTodos(
+        todos.map((todo) => {
+          if (todo.index === id) {
+            todo.title = inputValueEditing;
+          }
+          return todo;
+        })
+      );
+      setInputEditError(false);
+      setEditingItem(null);
+    } else {
+      setInputEditError(true);
+    }
+  }
+  return (
+    <>
+      <div className='add-item-wrapper'>
+        <input
+          type='text'
+          value={inputValue}
+          onChange={handleChangeInput}
+          onKeyDown={handleSubmitInputByEnter}
+          tabIndex={1}
+          className={inputError ? 'add-item__input error' : 'add-item__input'}
+          placeholder='Create a new ToDo&hellip;'
+        />
+        <Button title='add' className='button' onClick={addTask} />
+      </div>
+      <div className='input-error'>{showTextEror}</div>
+      <div className='status-wrapper'>
+        <Button
+          title='All'
+          className={
+            !!todos.length && filter === Filters.All ? 'button active-filter' : 'button'
+          }
+          onClick={() => setFilter(Filters.All)}
+          disabled={disableStatusButton}
+        />
+        <Button
+          title='Active'
+          className={!!todos.length && filter === Filters.Active ? 'button active-filter' : 'button'}
+          onClick={() => setFilter(Filters.Active)}
+          disabled={disableStatusButton}
+        />
+        <Button
+          title='Completed'
+          className={!!todos.length && filter === Filters.Completed ? 'button active-filter' : 'button'}
+          onClick={() => setFilter(Filters.Completed)}
+          disabled={disableStatusButton}
+        />
+      </div>
+      <div className='tasks'>
+        {!todos.length ? (
+          <div className='tasks__no-task'>There is no task to do&hellip;</div>
+        ) : (
+          todos
+            .filter((todo) => {
+              if (filter === Filters.All) {
+                return todos;
+              } else if (filter === Filters.Active) {
+                return !todo.isCompleted;
+              } else if (filter === Filters.Completed) {
+                return todo.isCompleted;
+              }
+            })
```

Comment:
Move to separate function 

Replace if/else condition to switch/case

Reply:
Done